01_financial-well-being/income_metrics_all_place.qmd

```diff
@@ -0,0 +1,622 @@
+
+```{r}
+
+if (file.exists(here::here("01_financial-well-being", "data", "raw", "income_1year.dat.gz")) == FALSE) 
+
+  {
+  
+ipums_files <-
+  list.files(paste0(here::here("01_financial-well-being", "data", "raw"))) %>%
+  as_tibble() %>%
+  filter(str_detect(value, "dat.gz|xml"),!str_detect(value, "income")) %>%
+  mutate(value = paste0(here::here("01_financial-well-being", "data", "raw"), "/", value)) %>%
+  pull()
+
+file.rename(ipums_files, c(
+  here::here("01_financial-well-being", "data", "raw", "income_1year.dat.gz"),
+  here::here("01_financial-well-being", "data", "raw", "income_1year.xml")
+))
+
+}
+```
+
+### Read in data
+
+Read in the micro data extract from the raw data folder.
+
+```{r}
+
+ddi <-
+  read_ipums_ddi(here::here("01_financial-well-being", "data", "raw", "income_1year.xml"))
+
+micro_data <-
+  read_ipums_micro(
+    ddi,
+    data_file = here::here("01_financial-well-being", "data", "raw", "income_1year.dat.gz")
+  )
+
+
+```
+
+Reformat variable names to lowercase for easier reference throughout the program.
+
+```{r}
+
+acs_2021 <- micro_data %>%
+  rename_with(tolower) 
+
+```
+
+Drop labels and define variable formats. Household income takes an unknown format upon reading in the raw data, we redefine it as numeric.
+Provide leading zeros to statefip and PUMA so they match the crosswalk. 
+
+```{r}
+
+acs_2021 <- acs_2021 %>%
+  mutate(
+    across(where(is.labelled), ~ lbl_clean(.x)),
+    hhincome = as.numeric(hhincome),
+    statefip = sprintf("%0.2d", as.numeric(statefip)),
+    puma = sprintf("%0.5d", as.numeric(puma))
+  )
+
+```
+
+Missing data is reported in a variety of ways by IPUMS - listed below:
+
+HHINCOME is missing it is reported as "9999999".
+Vacancy is reported as "0". 
+
+```{r}
+acs_2021 %>% 
+  ggplot(aes(x = hhincome)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Household Income Variable Results") +
+  ylab("Density")
+
+acs_2021 %>% 
+  ggplot(aes(x = vacancy)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Vacancy Variable Results") +
+  ylab("Density")
+```
+
+Note there are no vacancy results by default of the structure of census data read in. Turn the missing/"not in universe" variable into NAs.
+
+```{r}
+
+acs2021clean <- acs_2021 %>%
+  mutate(hhincome = ifelse(hhincome == 9999999, NA_integer_,
+                          hhincome)
+  )
+
+```
+
+Look at distribution after adjustment. Some out liers will still exist but bunching around the missing value should not be present.
+
+```{r}
+
+acs2021clean %>% 
+  ggplot(aes(x = hhincome)) +
+  geom_density(color = "blue", fill = alpha("blue", 0.3)) +
+  theme_minimal() +
+  ggtitle("Household Income Variable Results") +
+  ylab("Density")
+
+```
+
+Remove group quarters, we want to keep only households (GQ 1 and 2).
+
+```{r}
+
+acs2021clean <- acs2021clean %>%
+  filter(gq < 3)
```

Comment:
What do you make of "Additional households under 2000 definition" for `gq == 5`?

Reply:
I can't find a good definition of what the change entails, it's a very small number of records though. I think we should include given the IPUMS definition clearly lists them as households.